dev/com.ibm.ws.cloudant/src/com/ibm/ws/cloudant/internal/CloudantService.java

```diff
@@ -597,4 +606,33 @@ protected void setSsl(ServiceReference<Object> ref) { // com.ibm.wsspi.ssl.SSLCo
     protected void unsetSsl(ServiceReference<Object> ref) {
         sslConfig.unsetReference(ref);
     }
+
+    /**
+     * Return the cloudant client object used for a particular username and password
+     *
+     * @param resAuth
+     * @param loginPropertyList
+     * @return CloudantClient object
+     * @throws Exception
+     */
+    public Object getCloudantClient(int resAuth, List<? extends ResourceInfo.Property> loginPropertyList) throws Exception {
+        AuthData containerAuthData = null;
+        if (resAuth == ResourceInfo.AUTH_CONTAINER) {
+            containerAuthData = getContainerAuthData(loginPropertyList);
+        }
+
+        String userName = containerAuthData == null ? (String) props.get(USERNAME) : containerAuthData.getUserName();
+        String password = null;
+        if (containerAuthData == null) {
+            SerializableProtectedString protectedPwd = (SerializableProtectedString) props.get(PASSWORD);
+            if (protectedPwd != null) {
+                password = String.valueOf(protectedPwd.getChars());
+                password = PasswordUtil.getCryptoAlgorithm(password) == null ? password : PasswordUtil.decode(password);
+            }
+        } else
+            password = String.valueOf(containerAuthData.getPassword());
+
+        final ClientKey key = new ClientKey(null, userName, password);
```

Comment:
The null parameter here is the class loader for the path where the Cloudant driver is loaded from the application.  I don't think the validators will work properly in combination with the load-from-app feature, although the latter is currently disabled, so it would be more a concern if attempting to enable it at a later point.